firebase-firestore/src/androidMain/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -114,6 +140,27 @@ actual class WriteBatch(val android: com.google.firebase.firestore.WriteBatch) {
             ?.let { encoded -> android.update(documentRef.android, encoded.toMap()) }
             .let { this }
 
+    @JvmName("updateFieldsExtended")
+    actual inline fun <reified T> update(
+        documentRef: DocumentReference,
+        strategy: SerializationStrategy<T>,
+        data: T,
+        encodeDefaults: Boolean,
+        vararg fieldsAndValues: Pair<String, Any?>
+    ): WriteBatch {
+        val serializedItem = encode(data, encodeDefaults) as Map<String, Any>
+
+        val serializedFieldAndValues = fieldsAndValues.takeUnless { fieldsAndValues.isEmpty() }
+            ?.map { (field, value) -> field to encode(value, encodeDefaults) }?.toMap()
+
+        val result = (serializedFieldAndValues?.let {
+            serializedItem.plus(it)
+        } ?: serializedItem)
+
+        return android.update(documentRef.android, result).let { this }
+    }
```

Comment:
Android specific impl of new added method, basically it combines the mapping of data and fieldsAndValues into one update query. Use case example is an update on a data class A with fields A.field1 and A.field2 where field1 value needs to be updated and field2 value needs to be deleted. In that case new value for field1 is included on the "data" and field2 in the fieldsAndValues. (Due to limitation of FieldValue.delete() or FieldValue.serverTimestamp() that can only by included in fieldsAndValues and not in the data class itself)

Reply:
c5e313c 6e74c62 c9e98ce 30fc496